### PR TITLE
Add support for parallel without matrix

### PR DESCRIPTION
**Pull request by repomaa**
this allows passing an integer to the parallel setting of a job as described in the gitlab ci docs

### CODE REVIEW

File: tests/test-cases/parallel-matrix/.gitlab-ci.yml [hunk 1]
```diff
@@ -25,9 +25,9 @@ test-job:
   needs: [build-job, pre-job]
   script:
     - echo "NAME:'${NAME}' TIER:'${TIER}' $CI_NODE_INDEX/$CI_NODE_TOTAL"
-    - test -f files/build-job__bar_.txt
-    - test -f files/build-job__bear_.txt
     - test -f files/build-job__foo_.txt
+    - test -f files/build-job__bar_.txt
```
Comment (firecow) [inline tests/test-cases/parallel-matrix/.gitlab-ci.yml:29]: I'm not sure, why this change is needed, can you elaborate on it, please? :smile:

File: tests/test-cases/parallel-matrix/integration.parallel-matrix.test.ts [hunk 2]
```diff
@@ -22,6 +22,17 @@ test("parallel-matrix <test-job>", async () => {
         chalk`{blueBright test-job [bar,pro]} {greenBright >} NAME:'bar' TIER:'pro' 6/7`,
         chalk`{blueBright test-job [beb]    } {greenBright >} NAME:'beb' TIER:'' 7/7`,
 
+        chalk`{black.bgGreenBright  PASS } {blueBright pre-job           }`,
```
Comment (firecow) [inline tests/test-cases/parallel-matrix/integration.parallel-matrix.test.ts:25]: This test should be unaffected by a the new `parallel: <integer>` feature
Why do we need to add extra expectations to this test in this PR?

Comment (repomaa) [reply tests/test-cases/parallel-matrix/integration.parallel-matrix.test.ts:25]: i can split this to another PR. The job was actually failing because of the typo above.

Comment (firecow) [reply tests/test-cases/parallel-matrix/integration.parallel-matrix.test.ts]: https://github.com/firecow/gitlab-ci-local/pull/867 @repomaa I've made a PR fixing the failing test. Good catch, btw :smile:

**Comment by firecow** [comment]
I'm closing this for now. Feel free to reopen, if you get some time to work more on it.
